### PR TITLE
Fix for multiselect

### PR DESCRIPTION
**Pull request by roidy**
#51

### CODE REVIEW

**Comment by roidy** [comment]
Wiki will need to be updated:-

<!DOCTYPE html><head></head>
multiselect | DialogSelect.xml | heading=string,options=item00 / item01 / item02 etc,separator=' / ',autoclose=integer,preselect=0 / 1 / 2
-- | -- | --

File: resources/lib/method.py [hunk 1]
```diff
@@ -152,6 +152,28 @@ def _get_preselected_items(string):
...
+    def _get_preselected_items(string):
+        if not string:
+            return [-1]
+        try:
+            return [int(string)]
+        except TypeError:
+            return [-1]
+        except ValueError:
+            pass
+        items = _split_items(kwargs.get('preselect') or '')
```
Comment (jurialmunkey) [inline resources/lib/method.py]: The `preselect` kwarg is the value passed as `string` so there is no need to get it again. More importantly though, you should be checking `options` kwarg here not the `preselect` as the point of this block of code is to check if the dialog list actually has items before we spend time looking for the preselect index numbers.

File: resources/lib/method.py [hunk 2]
```diff
@@ -152,6 +152,28 @@ def _get_preselected_items(string):
...
+        if not items:
+            return [-1]
+        if len(items) == 0:
+            return [-1]
+        selected = []
+        for i in items:
+            try:
+                selected.append(int(i))
```
Comment (jurialmunkey) [inline resources/lib/method.py]: This is fine **if** ints are passed. However, the basic select function also allows passing strings to match the listitems so we need to look for the index too. I've got something for this incoming.

File: resources/lib/method.py [hunk 3]
```diff
@@ -217,14 +239,14 @@ def _get_preselected_items(string):
...
+                ('autoclose', int, 0), ('preselect', _get_preselected_item, -1), ('useDetails', boolean, False), )
         },
         'multiselect': {
-            'func': dialog.select,
+            'func': dialog.multiselect,
             'params': (
                 ('heading', str, ''),
-                ('list', _split_items, ''),
-                ('autoclose', int, 0), ('preselect', _get_preselected_items, -1), ('useDetails', boolean, False), )
+                ('options', _split_items, ''),
```
Comment (jurialmunkey) [inline resources/lib/method.py:267]: Not a problem but just a note at how annoying that the kwarg name is not the same for multiselect as select!

**Review by jurialmunkey — COMMENTED** [review]
Thanks. This is a great start but a couple of problems I noticed (which I've got something incoming for).

The bigger problem though which I'm not certain of best way to address is what to do with the output -- a string of a list of items won't be much use to pass to Kodi builtins. So there's no point even calling the dialog (other than for testing) if you can't use the output. EDIT: Figured something out for that too.

**Comment by jurialmunkey** [comment]
This commit that I pushed will resolve my comments (and also moves into a class to make the code a bit clearer) https://github.com/jurialmunkey/script.skinvariables/pull/52/commits/d1402bb2718780b5bc1457e7b8a5b5cdd6fe90c0

~How are you planning on using the multiselect dialog though? Because I'm trying to think through best way to handle executebuiltin on the output.~

EDIT: Nevermind. I thought of a simple/elegant solution -- just iterate through and execute which ever is selected e.g.

`RunScript(script.skinvariables,run_dialog=multiselect,heading=Multiselect test,options=Apple / Orange / Banana,preselect=Orange,"executebuiltin=Notification(Index {x}, Value {v})")`

With latest commits this should show multiselect with Apple, Orange and Banana as options. Orange should be preselected. It will then notify in order the selections made as "Index 0, Value Apple" etc.

**Pull request merged**